repo_policy_compliance/github_client.py

```diff
@@ -53,7 +67,119 @@ def get() -> Github:
         raise_on_status=False,
         raise_on_redirect=False,
     )
-    return Github(auth=Token(github_token), retry=retry_config)
+    return Github(auth=auth, retry=retry_config)
+
+
+def _get_auth() -> Auth:
+    """Get a GitHub auth object.
+
+    Returns:
+        A GitHub auth object that is configured with a token from the environment.
+    """
+    github_token = os.getenv(GITHUB_TOKEN_ENV_NAME) or os.getenv(f"FLASK_{GITHUB_TOKEN_ENV_NAME}")
+    github_app_id = os.getenv(GITHUB_APP_ID_ENV_NAME) or os.getenv(
+        f"FLASK_{GITHUB_APP_ID_ENV_NAME}"
+    )
+    github_app_installation_id_str = os.getenv(GITHUB_APP_INSTALLATION_ID_ENV_NAME) or os.getenv(
+        f"FLASK_{GITHUB_APP_INSTALLATION_ID_ENV_NAME}"
+    )
+    github_app_private_key = os.getenv(GITHUB_APP_PRIVATE_KEY_ENV_NAME) or os.getenv(
+        f"FLASK_{GITHUB_APP_PRIVATE_KEY_ENV_NAME}"
+    )
+
+    _ensure_either_github_token_or_app_config(
+        github_token=github_token,
+        github_app_id=github_app_id,
+        github_app_installation_id_str=github_app_installation_id_str,
+        github_app_private_key=github_app_private_key,
+    )
+
+    auth: Auth
+    # we use asserts here to make mypy happy, we have already checked that the values are not None
+    # we know asserts are optimised away in production, so ignore bandit warnings
+    if github_app_id:
+        assert github_app_installation_id_str is not None  # nosec
+        assert github_app_private_key is not None  # nosec
+        auth = _get_github_app_installation_auth(
```

Comment:
Out of curiosity: What happens after a configuration error is raised here?

Reply:
The charm should go into a blocked state. If the application is run without the charm, the error will just bubble up and stop the running process.